packages/@react-spectrum/s2/src/Accordion.tsx

```diff
@@ -0,0 +1,397 @@
+export {_AccordionItem as AccordionItem};
+
+export interface AccordionHeaderProps extends UnsafeStyles, DOMProps {
+  /** The heading level of the accordion header.
+   * 
+   * @default 3
+   */
+  level?: number,
+  children: React.ReactNode
+}
+
+const headingStyle = style({
+  margin: 0
+});
+
+const buttonStyles = style({
+  ...focusRing(),
+  outlineOffset: -2,
+  font: 'heading',
+  color: {
+    default: 'neutral',
+    isDisabled: 'disabled'
+  },
+  fontWeight: 'bold',
+  fontSize: {
+    size: {
+      S: 'ui-lg',
+      M: 'ui-xl',
+      L: 'ui-3xl',
+      XL: 'ui-5xl'
+    }
+  },
+  lineHeight: 'ui',
+  display: 'flex',
+  alignItems: 'center',
+  paddingX: {
+    size: {
+      S: sizeValue(7),
+      M: sizeValue(11),
+      L: sizeValue(14),
+      XL: sizeValue(17)
+    }
+  },
+  paddingY: {
+    size: {
+      S: {
+        density: {
+          compact: sizeValue(2),
+          regular: sizeValue(7),
+          spacious: sizeValue(11)
+        }
+      },
+      M: {
+        density: {
+          compact: sizeValue(5),
+          regular: sizeValue(9),
+          spacious: sizeValue(13)
+        }
+      },
+      L: {
+        density: {
+          compact: sizeValue(8),
+          regular: sizeValue(11),
+          spacious: sizeValue(16)
+        }
+      },
+      XL: {
+        density: {
+          compact: sizeValue(8),
+          regular: sizeValue(12),
+          spacious: sizeValue(16)
+        }
+      }
+    }
+  },
+  gap: {
+    size: {
+      S: sizeValue(7),
+      M: sizeValue(11),
+      L: sizeValue(14),
+      XL: sizeValue(17)
+    }
+  },
+  height: 'full',
+  minHeight: 'control',
+  width: 'full',
+  backgroundColor: {
+    default: 'transparent',
+    isFocusVisible: 'transparent-black-100',
+    isHovered: 'transparent-black-100'
+  },
+  borderWidth: 0,
+  borderRadius: {
+    // Only rounded for keyboard focus and quiet hover.
+    default: 'none',
+    size: {
+      S: {
+        isFocusVisible: '[7px]',
+        isQuiet: {
+          isHovered: '[7px]',
+          isFocusVisible: '[7px]'
+        }
+      },
+      M: {
+        isFocusVisible: 'default',
+        isQuiet: {
+          isHovered: 'default',
+          isFocusVisible: 'default'
+        }
+      },
+      L: {
+        isFocusVisible: '[9px]',
+        isQuiet: {
+          isHovered: '[9px]',
+          isFocusVisible: '[9px]'
+        }
+      },
+      XL: {
+        isFocusVisible: 'lg',
+        isQuiet: {
+          isHovered: 'lg',
+          isFocusVisible: 'lg'
+        }
+      }
+    }
+  },
+  textAlign: 'start',
+  disableTapHighlight: true
+});
+
+const chevronStyles = style({
+  rotate: {
+    isRTL: 180,
+    isExpanded: 90
+  },
+  transitionDuration: '100ms',
+  transitionProperty: 'rotate',
+  '--iconPrimary': {
+    type: 'fill',
+    value: 'currentColor'
+  },
+  flexShrink: 0
+});
+
+function AccordionHeader(props: AccordionHeaderProps, ref: DOMRef<HTMLDivElement>) {
```

Comment:
Wondering if we will need to split out a separate component for the header vs the heading/button so you could add adjacent actions or if that's possible to achieve with the current api?